### ISSUE TITLE
Calling `Select` on `TextBox` in Skia GTK causes it to get focused

### ISSUE THREAD

**Issue by MartinZikmund** [unoplatform/uno#7643]
### Current behavior

In #7273 we fixed the behavior of `TextBox` selection on Skia to properly return `SelectionStart` and `SelectionEnd` values. Unfortunately, this change also causes the managed `TextBox` to get focused - which is a problem for some controls like `NumberBox` which perform selection as part of control initialization.

### Expected behavior

Selection should not depend on focus.

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio 2022, Visual Studio 2019

**Comment by Youssef1313** [comment]
SelectRegion on Skia doesn't work unless the control is focused. Do you think it's sensible to find the previously focused element, then focus TextBox and do selection, then focus the previously focused element again?

Alternatively, if there is a way different than SelectRegion, it should be better.

**Comment by MartinZikmund** [comment]
@Youssef1313 For now I worked around it by storing the requested selection range in separate fields when the control is not focused. In UWP the selection disappears when the input is focused, so it should not be a big issue. It is important though for `NumberBox` so that is the reason why I had to avoid this focus behavior... Unfocusing and focusing would also be tricky, as that could potentially break some logic in apps (e.g. some input form where on unfocused there is some kind of validation)
